ambry-store/src/main/java/com.github.ambry.store/LogSegment.java

```diff
@@ -29,25 +35,34 @@
  * transparently redirect operations if required.
  */
 class LogSegment implements Read, Write {
+  private static final short VERSION = 0;
+  private static final int VERSION_HEADER_SIZE = 2;
+  private static final int CAPACITY_HEADER_SIZE = 8;
+  private static final int CRC_SIZE = 8;
+
+  static final int HEADER_SIZE = VERSION_HEADER_SIZE + CAPACITY_HEADER_SIZE + CRC_SIZE;
+
   private final FileChannel fileChannel;
   private final File file;
   private final long capacityInBytes;
   private final String name;
   private final Pair<File, FileChannel> segmentView;
   private final StoreMetrics metrics;
+  private final long startOffset;
   private final AtomicLong endOffset;
   private final AtomicLong refCount = new AtomicLong(0);
 
   /**
-   * Creates a LogSegment abstraction.
+   * Creates a LogSegment abstraction with the given capacity.
    * @param name the desired name of the segment. The name signifies the handle/ID of the LogSegment and may be
    *             different from the filename of the {@code file}.
    * @param file the backing {@link File} for this segment.
    * @param capacityInBytes the intended capacity of the segment
    * @param metrics the {@link StoreMetrics} instance to use.
+   * @param writeHeader if {@code true}, headers are written that provide metadata about the segment.
    * @throws IOException if the file cannot be read or created
    */
-  LogSegment(String name, File file, long capacityInBytes, StoreMetrics metrics)
+  LogSegment(String name, File file, long capacityInBytes, StoreMetrics metrics, boolean writeHeader)
```

Comment:
I feel that we should also append totalCapacity of the LogSegment while serializing the file.